### ISSUE TITLE
training loss is not showed in the progress bar since 2.0

### ISSUE THREAD

**Issue by function2-llx** [Lightning-AI/pytorch-lightning#17251]
### Bug description

The 2.0 release refactors the progress bar by splitting training and validation. However, it also (silently) removes the training loss shown in the progress bar, which I cannot find mentioned in related issues and pull requests.

The [release note](https://github.com/Lightning-AI/lightning/releases/tag/2.0.0) also shows this change, but just ignores it without any explanation:

![image](https://user-images.githubusercontent.com/38486514/229270195-cca62d78-9d90-4cd7-91ea-35c8a1acdc67.png)

@awaelchli 

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

cc @awaelchli

**Comment by awaelchli** [comment]
@function2-llx It was removed intentionally. In 2.0+ you need to log it like so: 

```py
self.log('loss', loss, prog_bar=True)
```
[Release notes](https://lightning.ai/pages/releases/2.0.0/#logging-loss)

**Comment by function2-llx** [comment]
Hi @awaelchli , thanks for your answer, sorry that I missed it in the release note (due to my poor searching on the page). 

May I ask another question: if we want the previous behavior, i.e., displaying the running average of training loss on the progress bar, do we have to calculate it manually?

**Comment by mitchelldehaven** [comment]
+1 for being able to display the running average. I'm often using models that are quite large, so running small batches with several gradient accumulations. When the batch size is 4, information about that specific batch's loss is not very useful compared to a running average.

**Comment by marcmk6** [comment]
From release note: `We are removing this feature to favor speed over automation.`
This seems to me saying that the displayed loss enabled by `prog_bar=True` is no longer a moving average but a real-time per-step value.
I want to say the whole reason I use lightning instead of raw PyTorch is all the automated works it does... So if there's not even an option for obtaining a moving average, this is a bad feature change imo

**Comment by carmocca** [comment]
I opened this suggestion in torchmetrics: https://github.com/Lightning-AI/torchmetrics/issues/1737

If this existed. You'd be able to create your running average metric and `self.log()` it

**Comment by awaelchli** [comment]
TorchMetrics 1.0 now includes the new Running() metric. Here is an example:

```py
import os

import torch
from torch.utils.data import DataLoader, Dataset
from torchmetrics.wrappers import Running
from torchmetrics.aggregation import MeanMetric
from lightning.pytorch import LightningModule, Trainer
from pytorch.loggers import CSVLogger


class RandomDataset(Dataset):
    def __init__(self, size, length):
        self.len = length
        self.data = torch.randn(length, size)

    def __getitem__(self, index):
        return self.data[index]

    def __len__(self):
        return self.len


class BoringModel(LightningModule):
    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 2)
        self.running_loss = Running(MeanMetric(), window=10)

    def forward(self, x):
        return self.layer(x)

    def training_step(self, batch, batch_idx):
        loss = self(batch).sum() * 1e-10 + batch_idx
        self.running_loss(loss)
        self.log("train_loss", self.running_loss.compute(), on_step=True)
        return {"loss": loss}

    def configure_optimizers(self):
        return torch.optim.SGD(self.layer.parameters(), lr=0.1)


def run():
    train_data = DataLoader(RandomDataset(32, 64), batch_size=2)

    model = BoringModel()
    trainer = Trainer(
        default_root_dir=os.getcwd(),
        logger=CSVLogger("logs"),
        num_sanity_val_steps=0,
        max_epochs=2,
        enable_model_summary=False,
        log_every_n_steps=1,
    )
    trainer.fit(model, train_data)


if __name__ == "__main__":
    run()

```

I'm not sure why I need to call `metric.compute()` for it to work with logging on step. 🤔

**Comment by tshu-w** [comment]
Disturbing warnings are raised saying the user tries to call compute before the update during sanity check runs.

**Comment by awaelchli** [comment]
Opened an issue on torchmetrics: https://github.com/Lightning-AI/torchmetrics/issues/2264

**Comment by awaelchli** [comment]
The warning was addressed in https://github.com/Lightning-AI/torchmetrics/pull/2265
